### PR TITLE
feat: [sc-29879] Update MATIC -> POL, Klayth -> Kaia

### PR DESCRIPTION
**Pull request by tw-daniel**
Story details: https://app.shortcut.com/trust-wallet-1/story/29879

1. Polygon chain migrated from MATIC symbol to POL. So we need to do migration in our services as well.
2. Klayth did rebranding, so we need to update the name, explorer and type.

### CODE REVIEW

**Review by nikoskarakostas — DISMISSED** [review]
also raised in DM, just checking here as well. Are we positive we don't also want to change the POL name too? Concern is coming from https://github.com/trustwallet/assets/commit/cadeaebd22104a9d7df908db167f53fe6315016e where we changed the name as well.

File: types/token.go [hunk 1]
```diff
@@ -89,7 +89,7 @@ const (
 	KIP20           TokenType = "KIP20"
 	APTOS           TokenType = "APTOS"
 	MOONBEAM        TokenType = "MOONBEAM"
-	KLAYTN          TokenType = "KLAYTN"
+	KLAYTN          TokenType = "KAIA"
```
Comment (iCell) [inline types/token.go:92]: Don't forget to update the records in the db

**Review by iCell — DISMISSED** [review]

File: types/token.go (hunk 1 above)
Comment (ilkamo) [reply types/token.go:92]: @iCell just for curiosity, which db are you talking about?

Comment (gipshtw) [reply types/token.go:92]: i think he is talking about `backend-assets` db

**Pull request merged**